sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -2171,6 +2404,201 @@ def get_transceiver_loopback(self):
+        errored_frames_min_host_input{lane_num}        = FLOAT                  ; Errored frames minimum value for host input
+        errored_frames_max_host_input{lane_num}        = FLOAT                  ; Errored frames maximum value for host input
+        errored_frames_avg_host_input{lane_num}        = FLOAT                  ; Errored frames average value for host input
+        errored_frames_curr_host_input{lane_num}       = FLOAT                  ; Errored frames current value for host input
+
+        ;C-CMIS specific fields
+        biasxi{lane_num}                               = FLOAT                  ; modulator bias xi in percentage
+        biasxq{lane_num}                               = FLOAT                  ; modulator bias xq in percentage
+        biasxp{lane_num}                               = FLOAT                  ; modulator bias xp in percentage
+        biasyi{lane_num}                               = FLOAT                  ; modulator bias yi in percentage
+        biasyq{lane_num}                               = FLOAT                  ; modulator bias yq in percentage
+        biasyp{lane_num}                               = FLOAT                  ; modulator bias yq in percentage
+        cdshort{lane_num}                              = FLOAT                  ; chromatic dispersion, high granularity, short link in ps/nm
+        cdlong{lane_num}                               = FLOAT                  ; chromatic dispersion, high granularity, long link in ps/nm  
+        dgd{lane_num}                                  = FLOAT                  ; differential group delay in ps
+        sopmd{lane_num}                                = FLOAT                  ; second order polarization mode dispersion in ps^2
+        soproc{lane_num}                               = FLOAT                  ; state of polarization rate of change in krad/s
+        pdl{lane_num}                                  = FLOAT                  ; polarization dependent loss in db
+        osnr{lane_num}                                 = FLOAT                  ; optical signal to noise ratio in db
+        esnr{lane_num}                                 = FLOAT                  ; electrical signal to noise ratio in db
+        cfo{lane_num}                                  = FLOAT                  ; carrier frequency offset in Hz
+        txcurrpower{lane_num}                          = FLOAT                  ; tx current output power in dbm
+        rxtotpower{lane_num}                           = FLOAT                  ; rx total power in  dbm
+        rxsigpower{lane_num}                           = FLOAT                  ; rx signal power in dbm
+        ========================================================================
+        """
+        vdm_real_value_dict = dict()
+        vdm_raw_dict = self.get_vdm(self.vdm.VDM_REAL_VALUE)
```

Comment:
Shall we perform VDM freeze and unfreeze to update min/max/avg statistical values?

Reply:
@mihirpat1 that is correct